### ISSUE TITLE
Redirect URLs with trailing slashes

### ISSUE THREAD

**Issue by kennethormandy** [sintaxi/harp#73]
Example app:
```
app/
  +- harp.json
  +- public/
      +- index.jade
      +- 404.jade
      +- articles/
          +- the-title.md
          +- another-one.md
```

**Expected Behaviour**
Visiting `harpdev.io/articles/the-title/` redirects to `harpdev.io/articles/the-title`

**Actual Behaviour**
A 404 is returned

I believe this is intentional, but could be improved slightly. The current behaviour does make sense, since `the-title` is actually a file masquerading as a directory. I could get the behaviour I wanted by changing the app to this:
```
app/
  +- harp.json
  +- public/
      +- index.jade
      +- 404.jade
      +- articles/
          +- the-title/
              +- index.md
          +- another-one/
              +- index.md
```
…but having the individual files is so much cleaner, I’d like to avoid that if possible. Additionally, people might link to things with the trailing `/`, expecting it to work.

In summary, when I visit `harpdev.io/articles/the-title/`, if there is no directory named `the-title` with an `index` file inside, I think I should be redirected to `harpdev.io/articles/the-title`. Phew!

**Comment by victorsegall** [comment]
> If I publish to the Harp Platform with "strict": true does it run in strict mode there?
>> Maybe, but I don't think it matters much because an application developed in "strict mode" would work perfectly fine with harp server in "non-strict mode" because all of these additional things that harp does for you is a superset of how a regular web server works.

This is true for the hyperlinks example, but maybe not for other cases.

__Turning off _ignoreme__

On one hand, that default feature is said to be inflexible in #373. On another, disabling it as described here would correctly show that such paths will not be ignored by default on some other HTTP daemon. Would a good solution not be to solve #373 through satisfying the idea here, and also considering (in the case of Apache as destination for compile) an .htaccess output from the compile routines?

And on that note, routing of "arbitrary slug line" to "arbitrary slug line.html" is an extremely common pattern in .htaccess config, as well.

**Comment by sintaxi** [comment]
Harp is now Intelligently indifferent about the trailing slash. Closing this issue.
